lensless/mask.py

```diff
@@ -0,0 +1,321 @@
+import abc
+import progressbar
+import numpy as np
+import cv2 as cv
+from math import sqrt
+from perlin_numpy import generate_perlin_noise_2d
+from sympy.ntheory import quadratic_residues
+from scipy.signal import max_len_seq
+from waveprop.fresnel import fresnel_conv
+from waveprop.rs import angular_spectrum
+
+
+
+
+
+class Mask(abc.ABC):
+    """
+    Parent Mask class
+    """
+
+    def __init__(self, sensor_size_px: tuple, 
+                 sensor_size_m: tuple, 
+                 feature_size: tuple, 
+                 distance_sensor: float, 
+                 wavelength: float) -> None:
+        """
+        Parent mask contructor.
+        Attributes common to each type of mask.
+
+        Parameters
+        ----------
+        sensor_size_px: tuple (dim=2)
+            size of the sensor (px)
+        sensor_size_m: tuple (dim=2)
+            size of the sensor (m)
+        feature_size: float
```

Comment:
float or tuple? maybe someone wants a rectangular shaped feature size

Reply:
It's the d1 I use in the waveprop functions

Reply:
I don't know if it's meant to be the pixel size

Reply:
put as TODO support for tuple, i.e. for angular spectrum or other propagation methods